packages/@react-aria/landmark/src/useLandmark.ts

```diff
@@ -0,0 +1,245 @@
+    }
+
+    // Wrap if necessary
+    if (nextLandmarkIndex < 0) {
+      nextLandmarkIndex = this.landmarks.length - 1;
+    } else if (nextLandmarkIndex >= this.landmarks.length) {
+      nextLandmarkIndex = 0;
+    }
+
+    return this.landmarks[nextLandmarkIndex];
+  }
+
+  /**
+   * Look at next landmark. If an element was previously focused inside, restore focus there.
+   * If not, focus the first focusable element inside the lanemark.
+   * If no focusable elements inside, go to the next landmark.
+   * If no landmarks at all, or none with focusable elements, don't move focus.
+   */
+  public f6Handler(e: KeyboardEvent) {
+    if (e.key === 'F6') {
+      e.preventDefault();
+      e.stopPropagation();
+
+      let backward = e.shiftKey;
+      let nextLandmark = this.getNextLandmark(e.target as HTMLElement, {backward});
+
+      // If no landmarks, return
+      if (!nextLandmark) {
+        return;
+      }
+
+      // Iterate through landmarks, starting at next landmark
+      // until we reach a landmark with a focusble element.
+      let initialNextLandmark = nextLandmark;
+      do {
+        // If something was previously focused in the next landmark, then return focus to it
+        if (nextLandmark.lastFocused) {
+          let lastFocused = nextLandmark.lastFocused;
+          if (document.body.contains(lastFocused)) {
+            lastFocused.focus();
+            return;
+          }
+        }
+
+        // Otherwise, focus the first focusable element in the next landmark
+        let leadingSentinal = nextLandmark.ref.current.previousSibling || nextLandmark.ref.current.parentElement;
+        // If we want to add a scope, this is what we're thinking:
+        // let trailingSentinal = nextLandmark.ref.current.nextSibling;
+        // let scope = [leadingSentinal, trailingSentinal];
+        let walker = getFocusableTreeWalker(nextLandmark.ref.current, {tabbable: true});
+        let nextNode = walker.nextNode() as HTMLElement;
+        if (!nextNode) {
+          walker.currentNode = leadingSentinal;
+          nextNode = walker.nextNode() as HTMLElement;
+        }
+        while (nextNode && this.closestLandmark(nextNode) !== nextLandmark) {
+          nextNode = walker.nextNode() as HTMLElement;
+        }
+        if (document.body.contains(nextNode)) {
+          nextNode.focus();
+          return;
+        }
+        
+        nextLandmark = this.getNextLandmark(nextLandmark.ref.current, {backward});
+      } while (nextLandmark !== initialNextLandmark);
+     
+    }
+  }
+
+  /**
+   * Sets lastFocused for a landmark, if focus is moved within that landmark.
+   */
+  public focusinHandler(e: FocusEvent) {
+    let currentLandmark = this.closestLandmark(e.target as HTMLElement);
+    this.landmarks = this.landmarks.map((landmark) => ({...landmark, ...(landmark === currentLandmark && {lastFocused: e.target as HTMLElement})}));
```

Comment:
We can use updateLandmark to handle this?
```suggestion
    this.updateLandmark({ref: currentLandmark, lastFocused: e.target as HTMLElement})
```

Reply:
Good idea, done